common/htlc_wire.c

```diff
@@ -147,6 +148,9 @@ void fromwire_bitcoin_tx(const u8 **cursor, size_t *max, struct bitcoin_tx *tx)
 	 * for the sake of simple structures, we don't write the
 	 * generator that way. */
 	struct bitcoin_tx *tx2 = pull_bitcoin_tx(tx, cursor, max);
-	if (tx2)
+	if (tx2) {
 		*tx = *tx2;
+		/* This hangs around with tx until freed */
+		notleak(tx2);
```

Comment:
Any reason we aren't just `tal_free`ing this one?

Reply:
Yes, because it has children we need.  We could steal all the children onto tx, then free, but this seemed th e lesser evil.